src/syft/core/io/route.py

```diff
@@ -119,9 +119,22 @@ def send_msg_without_reply(self, msg:SyftMessageWithoutReply) -> None:
 
 class BroadcastRoute(Route):
 
-    def send_msg_with_reply(self, msg:SyftMessageWithReply) -> Set[SyftMessageWithoutReply]:
-        raise NotImplementedError
+    def __init__(self, topic: str, connection: ClientConnection):
+        self.connection = connection
+        self.topic = topic
+
+    def send_msg_with_reply(self, msg):
```

Comment:
add type hints for args and return type. I don't know what you're returning from this unless it's specified.